src/sql/src/parsers/alter_parser.rs

```diff
@@ -41,7 +43,25 @@ impl<'a> ParserContext<'a> {
             } else {
                 let _ = parser.parse_keyword(Keyword::COLUMN);
                 let column_def = parser.parse_column_def()?;
-                AlterTableOperation::AddColumn { column_def }
+                let location = if parser.parse_keyword(Keyword::FIRST) {
+                    Some(AddColumnLocation::First)
+                } else if let Token::Word(word) = parser.peek_token().token {
+                    if word.value.to_uppercase() == "AFTER" {
+                        parser.next_token();
+                        let name = parser.parse_identifier()?;
+                        Some(AddColumnLocation::After {
+                            column_name: name.value,
+                        })
+                    } else {
+                        None
```

Comment:
I just try to get a location info here, if token existed but neither `FIRST` nor `AFTER`, then no location info can be parsed.

Token only be skipped if it is `FIRST` or `AFTER`, otherwise it will be kept and handled outside this function if it's unexpected, just the same as before.

![image](https://github.com/GreptimeTeam/greptimedb/assets/9639449/ec288309-e532-4ea7-9849-7489ab228e75)

Will add a sqlness test for this case.